From: dann frazier <dannf@debian.org>
To: submit@bugs.debian.org
Subject: hobbit: ldap_init, ldap_get_values implicitly converted to pointer
Date: Thu, 31 Jan 2008 08:52:40 -0700

Package: hobbit
Version: 4.2.0.dfsg-9
Severity: important
Tags: patch
Usertags: implicit-pointer-conversion

Our automated buildd log filter[1] detected a problem that is likely to
cause your package to segfault on architectures where the size of a
pointer is greater than the size of an integer, such as ia64 and amd64.

This is often due to a missing function prototype definition.
For more information, see [2].

[1] http://people.debian.org/~dannf/check-implicit-pointer-functions
[2] http://wiki.debian.org/ImplicitPointerConversions

  Function `ldap_init' implicitly converted to pointer at test-ldap.c:16
  Function `ldap_init' implicitly converted to pointer at ldaptest.c:140
  Function `ldap_get_values' implicitly converted to pointer at ldaptest.c:309

The libldap API has been updated and many functions used by the ldap
plugin are now deprecated. This package should either update to the
new API or define LDAP_DEPRECATED to continue using the deprecated
interfaces.

This patch implements the lazy solution.

diff -urpN hobbit-4.2.0.dfsg.orig/bbnet/ldaptest.h hobbit-4.2.0.dfsg/bbnet/ldaptest.h
--- hobbit-4.2.0.dfsg.orig/bbnet/ldaptest.h	2006-08-09 14:09:56.000000000 -0600
+++ hobbit-4.2.0.dfsg/bbnet/ldaptest.h	2008-01-31 08:47:12.000000000 -0700
@@ -17,6 +17,7 @@
 
 #ifdef BBGEN_LDAP
 #include <lber.h>
+#define LDAP_DEPRECATED 1
 #include <ldap.h>
 
 #ifndef LDAPS_PORT
diff -urpN hobbit-4.2.0.dfsg.orig/build/test-ldap.c hobbit-4.2.0.dfsg/build/test-ldap.c
--- hobbit-4.2.0.dfsg.orig/build/test-ldap.c	2006-08-09 14:09:57.000000000 -0600
+++ hobbit-4.2.0.dfsg/build/test-ldap.c	2008-01-31 08:46:33.000000000 -0700
@@ -2,6 +2,7 @@
 #include <stdio.h>
 
 #include <lber.h>
+#define LDAP_DEPRECATED 1
 #include <ldap.h>
 
 LDAPURLDesc ludp;

